src/data/data1d/database.ts

```diff
@@ -36,19 +38,32 @@ export interface InitiateDatabaseResult {
   data: DatabaseNMREntry[];
   getSolvents: () => string[];
   search: (keywords?: string | string[]) => DatabaseNMREntry[];
+  searchByStructure: (molfile: string) => DatabaseNMREntry[];
 }
 
 export function initiateDatabase(
   databases: DatabaseNMREntry[],
   nucleus: string,
 ): InitiateDatabaseResult {
   const data = databases.filter((datum) => datum.nucleus === nucleus);
+  const moleculesDB = prepareMoleculesDB(data);
 
   const getSolvents = () => prepareGetSolvents(data);
   const search = (keywords: string | string[] = []) =>
     filter(data, { keywords });
+  const searchByStructure = (molfile: string) =>
+    prepareSearchByStructure(moleculesDB, molfile);
 
-  return { data, getSolvents, search };
+  return { searchByStructure, data, getSolvents, search };
+}
+
+function prepareSearchByStructure(
+  moleculesDB: MoleculesDB,
+  molFile: string,
+): DatabaseNMREntry[] {
+  const molecule = OCL.Molecule.fromMolfile(molFile);
```

Comment:
In the callback of the editor we receive back a molecule and we should not change it to a molfile and then recreate a molecule. We are loosing information during this process !!!!